src/lib/utils/simd/simd_2x64/simd_2x64.h

```diff
@@ -0,0 +1,182 @@
+/**
+* (C) 2022,2024 Jack Lloyd
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_SIMD_2X64_H_
+#define BOTAN_SIMD_2X64_H_
+
+#if defined(BOTAN_TARGET_SUPPORTS_SSSE3)
+   #include <tmmintrin.h>
+   #define BOTAN_SIMD_2X64_ISA_FUNC BOTAN_FUNC_ISA("ssse3")
+#endif
+
+#if defined(BOTAN_TARGET_SUPPORTS_NEON)
+   #include <arm_neon.h>
+   #define BOTAN_SIMD_2X64_ISA_FUNC /**/
+#endif
+
+#include <botan/internal/cpuid.h>
+
+namespace Botan {
+
+class SIMD_2x64 final {
+   public:
+#if defined(BOTAN_TARGET_SUPPORTS_SSSE3)
+      using native_simd_type = __m128i;
+#elif defined(BOTAN_TARGET_SUPPORTS_NEON)
+      using native_simd_type = uint64x2_t;
+#endif
+
+      SIMD_2x64& operator=(const SIMD_2x64& other) = default;
+      SIMD_2x64(const SIMD_2x64& other) = default;
+
+      SIMD_2x64& operator=(SIMD_2x64&& other) = default;
+      SIMD_2x64(SIMD_2x64&& other) = default;
+
+      ~SIMD_2x64() = default;
+
+      // zero initialized
+      SIMD_2x64() {
+#if defined(BOTAN_TARGET_SUPPORTS_SSSE3)
+         m_simd = _mm_setzero_si128();
+#elif defined(BOTAN_TARGET_SUPPORTS_NEON)
+         m_simd = vdupq_n_u64(0);
+#endif
+      }
+
+      static SIMD_2x64 load_le(const void* in) {
+#if defined(BOTAN_TARGET_SUPPORTS_SSSE3)
+         return SIMD_2x64(_mm_loadu_si128(reinterpret_cast<const __m128i*>(in)));
+#elif defined(BOTAN_TARGET_SUPPORTS_NEON)
+         return SIMD_2x64(vld1q_u64(static_cast<const uint64_t*>(in)));
+#endif
+      }
+
+      void store_le(uint64_t out[2]) const { this->store_le(reinterpret_cast<uint8_t*>(out)); }
+
+      void store_le(uint8_t out[]) const {
+#if defined(BOTAN_TARGET_SUPPORTS_SSSE3)
+         _mm_storeu_si128(reinterpret_cast<__m128i*>(out), m_simd);
+#elif defined(BOTAN_TARGET_SUPPORTS_NEON)
+         vst1q_u8(out, vreinterpretq_u8_u64(m_simd));
+#endif
+      }
```

Comment:
Providing a `std::span<>` based overload (and additionally providing a `store_be`) will make this fulfill the `custom_storable` concept:

```suggestion
      void store_le(uint64_t out[2]) const { this->store_le({reinterpret_cast<uint8_t*>(out), sizeof(out)}); }

      void store_le(std::span<uint8_t, 16> out) const {
#if defined(BOTAN_TARGET_SUPPORTS_SSSE3)
         _mm_storeu_si128(reinterpret_cast<__m128i*>(out.data()), m_simd);
#elif defined(BOTAN_TARGET_SUPPORTS_NEON)
         vst1q_u8(out.data(), vreinterpretq_u8_u64(m_simd));
#endif
      }
```

Reply:
This doesn't compile and I don't see an obvious fix, I'm backing out this change.